commit 3acf41c5c9c538d22046b5f9822d58c9981e53da
Author: Greg Kroah-Hartman <gregkh@suse.de>
Date:   Tue Oct 12 13:18:17 2010 -0700

    Staging: brcm80211: include: remove unneeded use of uintptr
    
    uintptr shouldn't be used when you are just casting to a pointer anyway,
    so remove it from these usages.
    
    Cc: Brett Rudley <brudley@broadcom.com>
    Cc: Henry Ptasinski <henryp@broadcom.com>
    Cc: Nohee Ko <noheek@broadcom.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

--- a/drivers/staging/brcm80211/include/linux_osl.h
+++ b/drivers/staging/brcm80211/include/linux_osl.h
@@ -236,11 +236,11 @@ extern int osl_error(int bcmerror);
 			switch (sizeof(*(r))) { \
 			case sizeof(u8): \
 				__osl_v = \
-				readb((volatile u8*)((uintptr)(r)^3)); \
+				readb((volatile u8*)((r)^3)); \
 				break; \
 			case sizeof(u16): \
 				__osl_v = \
-				readw((volatile u16*)((uintptr)(r)^2)); \
+				readw((volatile u16*)((r)^2)); \
 				break; \
 			case sizeof(u32): \
 				__osl_v = readl((volatile u32*)(r)); \
@@ -255,10 +255,10 @@ extern int osl_error(int bcmerror);
 		switch (sizeof(*(r))) { \
 		case sizeof(u8):	\
 			writeb((u8)(v), \
-			(volatile u8*)((uintptr)(r)^3)); break; \
+			(volatile u8*)((r)^3)); break; \
 		case sizeof(u16):	\
 			writew((u16)(v), \
-			(volatile u16*)((uintptr)(r)^2)); break; \
+			(volatile u16*)((r)^2)); break; \
 		case sizeof(u32):	\
 			writel((u32)(v), \
 			(volatile u32*)(r)); break; \
--- a/drivers/staging/brcm80211/include/siutils.h
+++ b/drivers/staging/brcm80211/include/siutils.h
@@ -265,7 +265,7 @@ typedef struct si_info {
 	u32 oob_router;	/* oob router registers for axi */
 } si_info_t;
 
-#define	SI_INFO(sih)	(si_info_t *)(uintptr)sih
+#define	SI_INFO(sih)	(si_info_t *)sih
 
 #define	GOODCOREADDR(x, b) (((x) >= (b)) && ((x) < ((b) + SI_MAXCORES * SI_CORE_SIZE)) && \
 		IS_ALIGNED((x), SI_CORE_SIZE))
